allocate_test.go

```diff
@@ -6,7 +6,12 @@ import (
 
 func TestTx_allocatePageStats(t *testing.T) {
 	f := newFreelist()
-	f.ids = []pgid{2, 3}
+	if !f.IsFreelistTypeArray {
+		ids := []pgid{2, 3}
+		f.init(ids)
+	} else {
+		f.ids = []pgid{2, 3}
```

Comment:
actually you can also add a f.init func to the array type freelist. so we do not need this if/else anymore.

Reply:
changed by using `readIDs`